compiler/passes/src/type_checking/check_expressions.rs

```diff
@@ -632,6 +660,10 @@ impl<'a> ExpressionVisitor<'a> for TypeChecker<'a> {
                         .iter()
                         .zip(input.elements.iter())
                         .for_each(|(expected, expr)| {
+                            // Check that the component expression is not a tuple.
+                            if matches!(expr, Expression::Tuple(_)) {
+                                self.emit_err(TypeCheckerError::nested_tuple_expression(expr.span()))
```

Comment:
Here we ensure that tuples are not inside other tuples, but do we also check that tuples are not inside structs? I didn't see it in the code, but I may well have missed it.

Reply:
Answered below.